src/main/java/com/openshift/jenkins/plugins/pipeline/OpenShiftImageStreams.java

```diff
@@ -149,49 +153,66 @@ public SCMRevisionState calcRevisionsFromBuild(Run<?, ?> build,
     	String commitId = lastCommitId;
 			
 		ImageStreamRevisionState currIMSState = null;
+		if (commitId == null) {
+			commitId = this.getCommitId(listener, build.getEnvironment(listener));
+		}
+
 		if (commitId != null) {
 			currIMSState = new ImageStreamRevisionState(commitId);
 			listener.getLogger().println(String.format(MessageConstants.SCM_LAST_REV, currIMSState.toString()));
 		} else {
-	    	listener.getLogger().println(MessageConstants.SCM_NO_REV);
+			listener.getLogger().println(MessageConstants.SCM_NO_REV);
 		}
-		
+
 		
 		return currIMSState;
 	}
 
-	@Override
-	protected PollingResult compareRemoteRevisionWith(
-			AbstractProject<?, ?> project, Launcher launcher,
-			FilePath workspace, TaskListener listener, SCMRevisionState baseline)
-			throws IOException, InterruptedException {
-    	listener.getLogger().println(String.format(MessageConstants.SCM_COMP, DISPLAY_NAME, getImageStreamName(project.getEnvironment(null, listener)), tag, namespace));
-    	String commitId = this.getCommitId(listener, project.getEnvironment(null,listener));
-		
+	protected PollingResult compareRemoteRevisionInternal(EnvVars env, Launcher launcher, TaskListener listener, SCMRevisionState baseline) {
```

Comment:
OK cool/thanks.  Yeah, let's leave the UI pieces consistent and continue to share the common code where we don't attempt to leverage the PROJECT_NAME.

That said, the project name is also mounted in the file system at a well known location.  In hindsight I should have looked for it there (I do similar stuff with the token and cert in Auth.java).

We could add that as part of this pull, since we are mucking with this code path :-), and then not force the name space to be specified when setting the image stream pollers up.

Or if you'd rather just open an issue to handle this at a later time I'm good with that as well.